### ISSUE TITLE
Monte Carlo Cross Validation

### ISSUE THREAD

**Issue by msrepo** [naamiinepal/xrayto3D-benchmark#27]
Run k-fold Cross validation seems too many training runs and gives an unbiased estimate but with high variance. What is our next best option, given we want to run atmost 3 training runs per dataset per architecture?

Monte Carlo Cross Validation is an option. Cons: gives biased estimate but lower variance.
Correcting for bias in Monte Carlo Cross Validation

![image](https://github.com/naamiinepal/xrayto3D-benchmark/assets/10219364/f9574c8a-4032-4a2c-9bed-6479fe7ea154)
![image](https://github.com/naamiinepal/xrayto3D-benchmark/assets/10219364/42ce498c-2906-42f4-9449-d2e48e448958)
n1 training set, n2 test set, J such splits represents Monte Carlo Cross Validation
![image](https://github.com/naamiinepal/xrayto3D-benchmark/assets/10219364/4a891750-f8f1-4a4c-aa01-be32df868a5b)
taking many such splits (J larger) is good.
![image](https://github.com/naamiinepal/xrayto3D-benchmark/assets/10219364/461ab3c0-2e10-404a-8e1b-203cd8e26ddc)

in gist, using the 1st method: say we split 100 samples into 80 train and 20 test samples. We do this split 3 times(monte carlo CV) i.e. J = 3. Then, corrected variance = (1/J + n2/n1)*uncorrected variance = (1/3 + 1/4)*uncorrected variance.

![image](https://github.com/naamiinepal/xrayto3D-benchmark/assets/10219364/315ec1ee-cdd2-48e3-bdb6-5d5c59c9c66b)
![image](https://github.com/naamiinepal/xrayto3D-benchmark/assets/10219364/67eeb0ec-a24d-43af-86d1-f0878c4473d3)
The second method require modification, does not seem feasible here.

Relevant references:
[Nadeau and Bengio, 2003](http://www.iro.umontreal.ca/~lisa/bib/pub_subject/finance/pointeurs/techrep.pdf)
[Model Evaluation, Model Selection, and Algorithm Selection in Machine Learning](https://arxiv.org/abs/1811.12808)
